src/network/networkd-dhcp4.c

```diff
@@ -928,6 +928,10 @@ static int dhcp4_request_address(Link *link, bool announce) {
         if (r < 0)
                 return log_link_warning_errno(link, r, "DHCP error: no netmask: %m");
 
+        r = sd_dhcp_lease_get_server_identifier(link->dhcp_lease, &server);
+        if (r < 0)
+                return log_link_debug_errno(link, r, "DHCP error: failed to get DHCP server IP address: %m");
```

Comment:
Are there legitimate/uncontrollable reasons why this could fail? If so, given it's just for logging and doesn't affect functionality, what about continuing without it?

Reply:
Actually, the failure can be ignored. But this should not fail here. If this fails, then something is broken in sd-dhcp-client.c or sd-dhcp-lease.c.